aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -956,19 +994,58 @@ module aptos_framework::delegation_pool {
     public entry fun set_operator(
         owner: &signer,
         new_operator: address
-    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, CommissionPercentage {
         let pool_address = get_owned_pool_address(signer::address_of(owner));
         // synchronize delegation and stake pools before any user operation
         // ensure the old operator is paid its uncommitted commission rewards
         synchronize_delegation_pool(pool_address);
         stake::set_operator(&retrieve_stake_pool_owner(borrow_global<DelegationPool>(pool_address)), new_operator);
     }
 
+    /// Allows an owner to update the commission percentage for the operator of the underlying stake pool.
+    public entry fun update_commission_percentage(
+        owner: &signer,
+        new_commission_percentage: u64
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, CommissionPercentage {
+        assert!(new_commission_percentage <= MAX_FEE, error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE));
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        assert!(
+            operator_commission_percentage(pool_address) + MAX_COMMISSION_INCREASE >= new_commission_percentage,
+            error::invalid_argument(ETOO_LARGE_COMMISSION_INCREASE)
+        );
+        assert!(
+            stake::get_remaining_lockup_secs(pool_address) >= min_remaining_secs_for_commission_change(),
+            error::invalid_state(ETOO_LATE_COMMISSION_CHANGE)
+        );
+
+        // synchronize delegation and stake pools before any user operation
+        // ensure the operator is paid its uncommitted commission rewards
+        // with the old commission percentage
```

Comment:
+
// ensure the observed lockup cycle is up-to-date so that the new commission doesn't take effect this observed lockup, but rather the next one

Reply:
> ensure the observed lockup cycle is up-to-date

@alexfilip2 , could you elaborate more on this ^

Reply:
This comment is outdated now that the next commission becomes effective based on the lockup cycle's ending time.
However, it is still important that `synchronize_delegation_pool` is called firstly in order to make sure any pending commission change is applied, then `NextCommissionPercentage` is overridden with a new value.